src/Spryker/Zed/AppWebhook/Business/WebhookHandler/WebhookHandler.php

```diff
@@ -9,30 +9,73 @@
 
 use Generated\Shared\Transfer\WebhookRequestTransfer;
 use Generated\Shared\Transfer\WebhookResponseTransfer;
+use Spryker\Zed\AppWebhook\AppWebhookConfig;
+use Spryker\Zed\AppWebhook\Business\Exception\AllowedNumberOfRetriesExceededException;
+use Spryker\Zed\AppWebhook\Business\Identifier\IdentifierBuilderInterface;
+use Spryker\Zed\AppWebhook\Business\MessageBuilder\MessageBuilder;
+use Spryker\Zed\AppWebhook\Persistence\AppWebhookEntityManagerInterface;
+use Throwable;
 
 class WebhookHandler
 {
     /**
      * @param array<\Spryker\Zed\AppWebhook\Dependency\Plugin\WebhookHandlerPluginInterface> $webhookHandlerPlugins
      */
-    public function __construct(protected array $webhookHandlerPlugins)
-    {
+    public function __construct(
+        protected array $webhookHandlerPlugins,
+        protected AppWebhookConfig $appWebhookConfig,
+        protected AppWebhookEntityManagerInterface $appWebhookEntityManager,
+        protected IdentifierBuilderInterface $identifierBuilder
+    ) {
     }
 
     public function handleWebhook(WebhookRequestTransfer $webhookRequestTransfer, WebhookResponseTransfer $webhookResponseTransfer): WebhookResponseTransfer
     {
-        foreach ($this->webhookHandlerPlugins as $webhookHandlerPlugin) {
-            if (!$webhookHandlerPlugin->canHandle($webhookRequestTransfer)) {
-                continue;
-            }
+        // get the default identifier which will be only used when the webhook is not a retry
+        $identifier = $this->identifierBuilder->getIdentifier();
 
-            $webhookResponseTransfer = $webhookHandlerPlugin->handleWebhook($webhookRequestTransfer, $webhookResponseTransfer);
+        // Only persist the webhook if it's not a retry.
+        if ($webhookRequestTransfer->getIsRetry() !== true) {
+            $webhookRequestTransfer->setIdentifier($identifier);
+            $this->appWebhookEntityManager->saveWebhookRequest($webhookRequestTransfer);
         }
 
-        if ($webhookResponseTransfer->getIsSuccessful() === null) {
+        // Delete the webhook when the number of retries is exceeded and throw an exception.
+        if ($webhookRequestTransfer->getIsRetry() === true && $webhookRequestTransfer->getRetries() >= $this->appWebhookConfig->getAllowedNumberOfWebhookRetries()) {
+            $this->appWebhookEntityManager->deleteWebhookRequest($webhookRequestTransfer);
+
+            throw new AllowedNumberOfRetriesExceededException(MessageBuilder::allowedNumberOfRetriesExceeded());
```

Comment:
Add the logger here as we do in other places as well. https://spryker.atlassian.net/browse/ACP-3828